<launch>
    <param name="use_sim_time" value="false" />

    <!-- Main node -->
    <node name="orb_slam3" pkg="orb_slam3_ros" type="ros_mono" output="screen">
        <!-- change the topics according to the dataset -->
        <remap from="/camera/image_raw" to="/camera/image_raw"/>

        <!-- Parameters for original ORB-SLAM3 -->
        <param name="voc_file"      type="string" value="$(find orb_slam3_ros)/orb_slam3/Vocabulary/ORBvoc.txt.bin"/>
        <param name="settings_file" type="string" value="$(find orb_slam3_ros)/config/Monocular/EuRoC.yaml"/>

        <!-- Parameters for ROS -->
        <param name="world_frame_id"    type="string"   value="map" />
        <param name="cam_frame_id"      type="string"   value="camera" />
        <param name="enable_pangolin"   type="bool"     value="false" />
        
        <!-- my param -->
        <param name="xangle"    type="double"   value="9" />
        <param name="yangle"    type="double"   value="10" />
        <param name="zangle"    type="double"   value="100" />
        
    </node>
    
    <!-- Visualization -->
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find orb_slam3_ros)/config/orb_slam3_no_imu.rviz" output="screen" />
    
    <!-- Trajectory path -->
    <node pkg="hector_trajectory_server" type="hector_trajectory_server" name="trajectory_server_orb_slam3" output="screen" ns="orb_slam3_ros" >
        <param name="/target_frame_name" value="/world" />
        <param name="/source_frame_name" value="/camera" />
        <param name="/trajectory_update_rate" value="20.0" />
        <param name="/trajectory_publish_rate" value="20.0" />
    </node>
    
    
    <!-- pnt transformation-->
    <node name = "pnt_transform" type = "pnt_trans" pkg = "orb_slam3_ros">
        <param name="offline" value="false" />

    </node>


    <!-- octmap -->
    <node pkg="octomap_server" type="octomap_server_node" name="octomap_server">

    	<!-- resolution in meters per pixel -->
	<param name="resolution" value="0.06" />

    	<!-- name of the fixed frame, needs to be "/map" for SLAM -->
    	<param name="frame_id" type="string" value="map" />

    	<!-- max range / depth resolution of the kinect in meter -->
    	<param name="sensor_model/max_range" value="100" />
    	<param name="latch" value="false" />

    	<!-- max/min height for occupancy map, should be in meters -->
    	<param name="pointcloud_max_z" value="0.8" />
    	<param name="pointcloud_min_z" value="-0.9" />
    	<param name="filter_speckles" value="true" />

        <param name="sensor_model/miss" value="0.1" />
        <param name="sensor_model/hit" value="0.7" />

        <param name="senspr_model/max" value ="1" />
        <param name="sensor_model/min" value="0.4" />
    	<!-- topic from where pointcloud2 messages are subscribed -->
    	<remap from="/cloud_in" to="/transformed_points" />
 
    </node>    
    
</launch>

